src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java

```diff
@@ -1,44 +1,290 @@
+import org.springframework.http.MediaType;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.util.SocketUtils;
 
+import cucumber.api.java.After;
+import cucumber.api.java.Before;
 import cucumber.api.java.en.Given;
 import cucumber.api.java.en.Then;
 import cucumber.api.java.en.When;
+import gherkin.deps.com.google.gson.JsonArray;
+import gherkin.deps.com.google.gson.JsonElement;
+import gherkin.deps.com.google.gson.JsonParser;
 
 @Ignore
+@AutoConfigureMockMvc
 public class SubscriptionTriggerSteps extends FunctionalTestBase {
 
+    private static final String SUBSCRIPTION_WITH_JSON_PATH = "src/functionaltests/resources/subscription_multiple.json";
+    private static final String EIFFEL_EVENTS_JSON_PATH = "src/functionaltests/resources/eiffel_events_for_test.json";
+
+    private static final String REST_ENDPOINT = "/rest";
+    private static final String REST_ENDPOINT_AUTH = "/rest/with/auth";
+    private static final String REST_ENDPOINT_PARAMS = "/rest/with/params";
+    private static final String REST_ENDPOINT_AUTH_PARAMS = "/rest/with/auth/params";
+    private static final String BASE_URL = "localhost";
+    
+    private static ObjectMapper objectMapper = new ObjectMapper();
+    private List<String> subscriptionNames = new ArrayList<>();
+
+    @Value("${email.sender}")
+    private String sender;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private RmqHandler rmqHandler;
+
+    @Autowired
+    private JavaMailSenderImpl mailSender;
+
+    private MvcResult result;
+    private MvcResult postResult;
+    private MvcResult getResult;
+    private SimpleSmtpServer smtpServer;
+    private ClientAndServer restServer;
+    private MockServerClient mockClient;
+
     private static final Logger LOGGER = LoggerFactory.getLogger(SubscriptionTriggerSteps.class);
 
-    @Given("^Subscription rest trigger is created$")
-    public void subscription_rest_trigger_is_created() throws Throwable {
-        LOGGER.debug("Creating subscription rest trigger");
-        System.out.println("MongoDB port for " + this.getClass().getName() + " is: " + getMongoDbPort());
+    @Before("@SubscriptionTriggerScenario")
+    public void beforeScenario() {
+        setupSMTPServer();
+        setupRestEndpoints();
+    }
+
+    @After("@SubscriptionTriggerScenario")
+    public void afterScenario() {
+        LOGGER.debug("Stopping SMTP and REST Mock Servers");
+        try {
+            smtpServer.stop();
+            restServer.stop();
+            mockClient.close();
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
```

Comment:
If body of both catch blocks are same, it is better to merge them. Not only here. Check it in whole your changes.

Reply:
Done.